Tests/WorkspaceTests/WorkspaceTests.swift

```diff
@@ -560,10 +566,15 @@ final class WorkspaceTests: XCTestCase {
             XCTAssertEqual(try popen([Git.tool, "-C", editRepoPath.asString, "rev-parse", "--abbrev-ref", "HEAD"]).chomp(), "HEAD")
           #endif
 
-            do {
-                try workspace.edit(dependency: editedDependency, packageName: aManifest.name, revision: dependency.checkoutState!.revision)
-                XCTFail("Unexpected success, \(editedDependency) is already in edit mode")
-            } catch _ as WorkspaceDiagnostics.DependencyAlreadyInEditMode {}
+            workspace.edit(
+                dependency: editedDependency,
+                packageName: aManifest.name,
+                diagnostics: diagnostics,
+                revision: dependency.checkoutState!.revision)
+            XCTAssert(diagnostics.hasErrors)
+            XCTAssert(diagnostics.diagnostics.contains(where: {
+                $0.id == WorkspaceDiagnostics.DependencyAlreadyInEditMode.id
+            }))
```

Comment:
Off topic: We should get some assert helpers for diagnostics engine.
`XCTAssertNoErrors(diagnostics)`
`XCTAssertContains(diagnostics, WorkspaceDiagnostics.DependencyAlreadyInEditMode.id)`